pinot-query-planner/src/main/java/org/apache/pinot/query/parser/CalciteRexExpressionParser.java

```diff
@@ -210,6 +210,10 @@ private static Expression compileFunctionExpression(RexExpression.FunctionCall r
     for (RexExpression childNode : childNodes) {
       operands.add(toExpression(childNode, pinotQuery));
     }
+    // for COUNT, add a literal val 1 to operand list b/c V1 doesn't handle empty operand functions.
+    if (functionKind == SqlKind.COUNT && operands.isEmpty()) {
+      operands.add(RequestUtils.getLiteralExpression(1));
```

Comment:
To keep the existing V1 behavior, let's put `RequestUtils.getIdentifierExpression("*")`